lib/chore_runner/dsl.ex

```diff
@@ -52,6 +52,32 @@ defmodule ChoreRunner.DSL do
           bool: 1
         ]
 
+      def create_download(filename, opts) do
+        opts = Keyword.put_new_lazy(opts, :chore_id, fn -> Process.get(:chore_id) end)
+
+        filename
```

Comment:
I'd prefer this to be written as a `case` or `with` rather than using `tap` to handle flow control.

The `with` could be something like this
```elixir
with {:ok, download} <- ChoreRunner.Downloads.create_download(filename, opts),
      _ <- ChoreRunner.Reporter.register_download(download),
      pubsub when not is_nil(pubsub) <- Process.get(:chore_pubsub) do
  Phoenix.PubSub.broadcast(pubsub, ChoreRunner.downloads_pubsub_topic(), :downloads_updated)
else
  _ -> :noop
end
```

Reply:
Actually, disregard this. I just realized why it is using tap.